plugin-core/src/main/java/org/opennms/plugins/cloud/rest/CloudConfigRestServiceImpl.java

```diff
@@ -60,18 +65,17 @@ public Response putActivationKey(final String keyJson) {
 
     @Override
     public Response putDeactivateKey(final String keyJson) {
-        try {
-            String key = extractKey(keyJson);
-            this.cm.initConfiguration(key);
-            this.cm.configure();
-            this.cm.deactivateKeyConfiguration(key);
-            
-        } catch (Exception e) {
-            return Response
-                    .status(Response.Status.INTERNAL_SERVER_ERROR)
-                    .entity(exceptionToJson(e))
+        if (this.cm.getStatus() == AUTHENTCATED || this.cm.getStatus() == CONFIGURED) {
```

Comment:
I wonder what happens if status is FAILED.
what should happen if we deactivate plugin? I guess the status should still be DEACTIVATED?